PURL-SPECIFICATION.rst

```diff
@@ -259,14 +257,31 @@ These ``purl`` separator characters MUST NOT be percent-encoded when used as
 Percent-encoding rules
 ----------------------
 
-When applying percent-encoding or decoding to a string, use the rules of RFC
-3986 section 2 (https://datatracker.ietf.org/doc/html/rfc3986#section-2).
+Unless otherwise provided in this specification, when applying percent-encoding
+or decoding to a string, use the rules of RFC 3986 section 2
+(https://datatracker.ietf.org/doc/html/rfc3986#section-2).  In the event of any
+conflict between this specification and RFC 3986 section 2, this specification
+governs.
```

Comment:
Are we going back to how it was before? The way I understood the current version was that it was a breaking change to intentionally break from RFC3986 section 2 because the RFC3986 encoding rules are more complicated to implement and most PURL implementations did not try to implement them, instead applying mostly the same encoding rules to all components. I've already had somebody ask why phylum-dev/purl doesn't encode plus signs in version numbers, which isn't required by RFC 3968 and is non-canonical according to the old PURL but is required by the rules in the current version of PURL, or at least we both thought so.

Reply:
@matt-phylum can you elaborate? I am not sure I understand fully your point. I thing that what we are trying to convey is that:
1. this spec defines WHICH characters to encode and WHERE/WHEN (e.g., with specifics for separators and components)
2. we defer to RFC3986 to define HOW to encode characters we want encoded.

Would this be clearer?

Reply:
By saying "In the event of any conflict between this specification and RFC 3986 section 2" it sounds to me like the implementer is supposed to combine RFC 3986 and PURL rules, eg merging RFC 3986 pchar with the PURL character rules when outputting a package name. If the PURL spec is clearly specifying WHICH characters WHERE/WHEN and RFC 3986 is specifying HOW then it's much easier to implement and there shouldn't be conflicts.